custom_reg_form/management/commands/email_opt_in_list_to_s3.py

```diff
@@ -0,0 +1,257 @@
+                return model_class
+        except Exception:   # pylint: disable=broad-except
+            # we ignore execptions so we can try to get the
+            # model class in another way
+            pass
+
+    # check REGISTRATOIN_EXTENSION_MODEL setting
+    if hasattr(settings, "REGISTRATION_EXTENSION_MODEL"):
+        try:
+            model_class_path = settings.REGISTRATION_EXTENSION_MODEL
+            module_path = ".".join(model_class_path.split(".")[:-1])
+            module = importlib.import_module(module_path)
+            model_class = getattr(module, model_class_path.split(".")[-1])
+            return model_class
+        except Exception:    # pylint: disable=broad-except
+            # ignore and try again
+            pass
+
+    # check REGISTRATION_EXTENSION_MODEL environment variable
+    if "REGISTRATION_EXTENSION_MODEL" in os.environ:
+        try:
+            model_class_path = os.environ.get("REGISTRATION_EXTENSION_MODEL")
+            module_path = ".".join(model_class_path.split(".")[:-1])
+            module = importlib.import_module(module_path)
+            model_class = getattr(module, model_class_path.split(".")[-1])
+            return model_class
+        except Exception:  # pylint: disable=broad-except
+            pass
+
+    raise Exception("Couldn't infer the model path to the registration extension model.")
+
+def infer_field_model(field):
+    """Returns the model of which `field` is a member"""
+
+    # first check in the User model
+    if hasattr(User, field):
+        return User
+    # check if in the UserProfile model
+    if hasattr(UserProfile, field):
+        return UserProfile
+    # check if in the REGISTRAION_EXTENSION_MODEL
+    reg_ext_model_class = get_registration_ext_model()
+    if hasattr(reg_ext_model_class, field):
+        return reg_ext_model_class
+
+    raise Exception("Couldn't infer field: `{}` model.".format(field))
+
+def construct_query(fields):
+    """Returns a string `query` to be passed to `cursor.execute`"""
+    reg_ext_table = get_registration_ext_model()._meta.db_table # pylint: disable=protected-access
+    field_names = ", ".join(
+        [".".join([infer_field_model(field)._meta.db_table, field]) for field in fields]    # pylint: disable=protected-access
+    )
+    query = """
+        SELECT 
+            {field_names}
+        FROM auth_user 
+            LEFT JOIN {reg_ext_table} ON {reg_ext_table}.user_id = auth_user.id
+            JOIN auth_userprofile ON auth_userprofile.user_id = auth_user.id;
+        """.format(
+            field_names=field_names, reg_ext_table=reg_ext_table
+        )
+
+    return query
+
+
+class Command(BaseCommand):
+    """
+    Admin command to export User/UserProfile/Registration Extension fields to csv file on s3
+    """
+
+    help = "Exports users email opt-in preferences to an s3 bucket"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--use-temp-file",
```

Comment:
This is missing from the usage example above.

Reply:
I believe it's description is well documented from the help command the usage section was used to document the environment variables and settings used, and how the fields are retrieved.